client/components/select-dropdown/test/index.js

```diff
@@ -0,0 +1,317 @@
+
+			sinon.assert.calledOnce( setStateSpy );
+			sinon.assert.calledWith( setStateSpy, { isOpen: false } );
+
+			expect( fakeContext.focused ).to.be.undefined;
+		} );
+	} );
+
+	describe( 'navigateItem', function() {
+		it( 'permits to navigate through the dropdown\'s options by pressing the TAB key', function() {
+			const tabKeyCode = 9;
+			const fakeEvent = prepareFakeEvent( tabKeyCode );
+			const fakeContext = prepareFakeContext();
+
+			SelectDropdown.prototype.navigateItem.call( fakeContext, fakeEvent );
+
+			sinon.assert.calledOnce( fakeContext.navigateItemByTabKey );
+			sinon.assert.calledWith( fakeContext.navigateItemByTabKey, fakeEvent );
+		} );
+
+		it( 'permits to select an option by pressing ENTER, or SPACE', function() {
+			function runNavigateItemTest( keyCode ) {
+				const fakeEvent = prepareFakeEvent( keyCode );
+				const fakeContext = prepareFakeContext();
+
+				SelectDropdown.prototype.navigateItem.call( fakeContext, fakeEvent );
+
+				sinon.assert.calledOnce( fakeEvent.preventDefault );
+				sinon.assert.calledOnce( fakeContext.activateItem );
+			}
+
+			const enterKeyCode = 13;
+			const spaceKeyCode = 32;
+
+			[ enterKeyCode, spaceKeyCode ].forEach( runNavigateItemTest );
+		} );
+
+		it( 'permits to close the dropdown by pressing ESCAPE', function() {
+			const escapeKeyCode = 27;
+			const fakeEvent = prepareFakeEvent( escapeKeyCode );
+			const fakeContext = prepareFakeContext();
+
+			SelectDropdown.prototype.navigateItem.call( fakeContext, fakeEvent );
+
+			sinon.assert.calledOnce( fakeEvent.preventDefault );
+
+			const {
+				refs: { dropdownContainer: { focus: focusSpy } },
+				closeDropdown: closeDropdownSpy
+			} = fakeContext;
+			sinon.assert.calledOnce( closeDropdownSpy );
+			sinon.assert.calledOnce( focusSpy );
+		} );
+
+		it( 'permits to open the dropdown, and navigate through the dropdown\'s options by pressing the arrow UP/DOWN keys', function() {
+			function runNavigateItemTest( { keyCode, direction } ) {
+				const fakeEvent = prepareFakeEvent( keyCode );
+				const fakeContext = prepareFakeContext();
+
+				SelectDropdown.prototype.navigateItem.call( fakeContext, fakeEvent );
+
+				sinon.assert.calledOnce( fakeEvent.preventDefault );
+
+				sinon.assert.calledOnce( fakeContext.focusSibling );
+				sinon.assert.calledWith( fakeContext.focusSibling, direction );
+
+				sinon.assert.calledOnce( fakeContext.openDropdown );
+			}
+
+			const arrowUp = { keyCode: 38, direction: 'previous' };
+			const arrowDown = { keyCode: 40, direction: 'next' };
+
+			[ arrowUp, arrowDown ].forEach( runNavigateItemTest );
+		} );
+
+		function prepareFakeContext() {
+			const activateItemSpy = sinon.spy();
+			const closeDropdownSpy = sinon.spy();
+			const focusSiblingSpy = sinon.spy();
+			const navigateItemByTabKeySpy = sinon.spy();
+			const openDropdownSpy = sinon.spy();
+			return {
+				refs: {
+					dropdownContainer: {
+						focus: sinon.spy()
+					}
+				},
+				activateItem: activateItemSpy,
```

Comment:
You can inline `sinon.spy` calls to make it more concise here and in `prepareFakeEvent`.